source/module_base/test/assoc_laguerre_test.cpp

```diff
@@ -0,0 +1,308 @@
+    for(int i = 0; i < xs.size(); i++)
+    {
+        EXPECT_NEAR(al.associate_laguerre(n, xs[i], k), ref_ys[i], 1e-6);
+    }
+    // test n = 8, k = 2
+    n = 8;
+    k = 2;
+    xs = {0.0, 1.0, 2.0, 3.0};
+    ref_ys = {45.0, -4.189459325396824, 0.7523809523809523, 3.1359374999999994};
+    for(int i = 0; i < xs.size(); i++)
+    {
+        EXPECT_NEAR(al.associate_laguerre(n, xs[i], k), ref_ys[i], 1e-6);
+    }
+    // test n = 8, k = 3
+    n = 8;
+    k = 3;
+    xs = {0.0, 1.0, 2.0, 3.0};
+    ref_ys = {165.0, 1.449231150793654, -6.384126984126984, 5.70200892857143};
+    for(int i = 0; i < xs.size(); i++)
+    {
+        EXPECT_NEAR(al.associate_laguerre(n, xs[i], k), ref_ys[i], 1e-6);
+    }
+    // test n = 8, k = 4
+    n = 8;
+    k = 4;
+    xs = {0.0, 1.0, 2.0, 3.0};
+    ref_ys = {495.0, 51.8809771825397, -20.098412698412695, -0.09441964285714727};
+    for(int i = 0; i < xs.size(); i++)
+    {
+        EXPECT_NEAR(al.associate_laguerre(n, xs[i], k), ref_ys[i], 1e-6);
+    }
+    // test n = 8, k = 5
+    n = 8;
+    k = 5;
+    xs = {0.0, 1.0, 2.0, 3.0};
+    ref_ys = {1287.0, 242.01411210317463, -21.990476190476176, -23.028348214285717};
+    for(int i = 0; i < xs.size(); i++)
+    {
+        EXPECT_NEAR(al.associate_laguerre(n, xs[i], k), ref_ys[i], 1e-6);
+    }
+    // test n = 8, k = 6
+    n = 8;
+    k = 6;
+    xs = {0.0, 1.0, 2.0, 3.0};
+    ref_ys = {3003.0, 777.6319692460318, 53.67301587301589, -60.999776785714275};
+    for(int i = 0; i < xs.size(); i++)
+    {
+        EXPECT_NEAR(al.associate_laguerre(n, xs[i], k), ref_ys[i], 1e-6);
+    }
+    // test n = 8, k = 7
+    n = 8;
+    k = 7;
+    xs = {0.0, 1.0, 2.0, 3.0};
+    ref_ys = {6435.0, 2055.2262152777776, 368.62539682539676, -75.53370535714284};
+    for(int i = 0; i < xs.size(); i++)
+    {
+        EXPECT_NEAR(al.associate_laguerre(n, xs[i], k), ref_ys[i], 1e-6);
+    }
+    // test n = 8, k = 8
+    n = 8;
+    k = 8;
+    xs = {0.0, 1.0, 2.0, 3.0};
+    ref_ys = {12870.0, 4777.330183531744, 1256.2666666666667, 53.21986607142856};
+    for(int i = 0; i < xs.size(); i++)
+    {
+        EXPECT_NEAR(al.associate_laguerre(n, xs[i], k), ref_ys[i], 1e-6);
+    }
+    // all physical and possible n, k values are tested
+}
+
+TEST_F(AssocLaguerreTest, FactorialTest)
+{
+    // this test is simple, no need to compare with reference value
+    Assoc_Laguerre al;
+    EXPECT_DOUBLE_EQ(al.factorial(0), 1);
+    EXPECT_DOUBLE_EQ(al.factorial(1), 1);
+    EXPECT_DOUBLE_EQ(al.factorial(2), 2);
+    EXPECT_DOUBLE_EQ(al.factorial(3), 6);
+    EXPECT_DOUBLE_EQ(al.factorial(4), 24);
+    EXPECT_DOUBLE_EQ(al.factorial(5), 120);
+    EXPECT_DOUBLE_EQ(al.factorial(6), 720);
+    EXPECT_DOUBLE_EQ(al.factorial(7), 5040);
+    EXPECT_DOUBLE_EQ(al.factorial(8), 40320);
+    EXPECT_DOUBLE_EQ(al.factorial(9), 362880);
+    EXPECT_DOUBLE_EQ(al.factorial(10), 3628800);
+}
+
+TEST_F(AssocLaguerreTest, ValueTest)
```

Comment:
Associated Laguerre series generation is tested segment-by-segment, each segment has different number of grids